### PR TITLE
drive a wooden stake through the heart of SSLv2Hello protocol [DPP-1218]

### PR DESCRIPTION
**Pull request by mziolekda**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/ProtocolDisabler.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      .appendToProperty(disabledAlgorithmsProperty, sslV2Protocol)
+}
+
+private[tls] case class PropertiesUpdater(
+    getter: String => String,
+    setter: (String, String) => Unit,
+) {
+  def appendToProperty(name: String, value: String): Unit = {
+    val property = getter(name)
+    val fullVal = value.r.findFirstIn(property).map(_ => property).getOrElse(s"$property, $value")
```
Comment (simonmaxen-da) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/ProtocolDisabler.scala:26]: Maybe better to split based on comma and then reconstruct string with `mkString`.  The reason is that the value may exist as a substring of another algorithm (e.g. DES and 3DES_EDE_CBC).

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/TlsConfiguration.scala [hunk 2]
```diff
@@ -116,17 +116,30 @@ final case class TlsConfiguration(
...
     logger.debug(
       s"$who TLS $who - supported cipher suites: ${tlsInfo.supportedCipherSuites.mkString(", ")}."
     )
     logger.info(s"$who TLS - enabled cipher suites: ${tlsInfo.enabledCipherSuites.mkString(", ")}.")
   }
 
   /** This is a side-effecting method. It modifies JVM TLS properties according to the TLS configuration. */
-  def setJvmTlsProperties(): Unit =
+  def setJvmTlsProperties(): Unit = {
     if (enabled && enableCertRevocationChecking) OcspProperties.enableOcsp()
```
Comment (simonmaxen-da) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/TlsConfiguration.scala:133]: May be nicer to check for enabled once.

**Review by simonmaxen-da — APPROVED** [review]

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/ProtocolDisabler.scala (hunk 1 above)
Comment (mziolekda) [reply ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/ProtocolDisabler.scala:26]: Fair point!

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/TlsConfiguration.scala (hunk 2 above)
Comment (mziolekda) [reply ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/TlsConfiguration.scala:133]: :+1:

**Pull request merged**